eitprocessing/eit_data/__init__.py

```diff
@@ -217,78 +225,76 @@ def _sliced_copy(
         end_index: int,
         label: str,
     ) -> Self:
-        cls = self._get_vendor_class(self.vendor)
+        cls = self.__class__
         time = self.time[start_index:end_index]
         nframes = len(time)
 
         phases = list(filter(lambda p: start_index <= p.index < end_index, self.phases))
         events = list(filter(lambda e: start_index <= e.index < end_index, self.events))
 
-        obj = cls(
+        pixel_impedance = self.pixel_impedance[start_index:end_index, :, :]
+
+        return cls(
             path=self.path,
             nframes=nframes,
             time=time,
             framerate=self.framerate,
             phases=phases,
             events=events,
             label=label,
+            pixel_impedance=pixel_impedance,
         )
 
-        for variant in self.variants.values():
-            obj.variants.add(
-                variant._sliced_copy(  # pylint: disable=protected-access
-                    start_index, end_index
-                )
-            )
+    def __len__(self):
+        return self.pixel_impedance.shape[0]
 
-        return obj
+    @property
+    def global_baseline(self) -> np.ndarray:
+        return np.nanmin(self.pixel_impedance)
 
-    @classmethod
-    @abstractmethod
-    def _from_path(  # pylint: disable=too-many-arguments
-        cls: type[Self],
-        path: Path,
-        label: str | None = None,
-        framerate: float | None = None,
-        first_frame: int | None = None,
-        max_frames: int | None = None,
-        return_non_eit_data: bool = False,
-    ) -> Self | tuple[Self, ContinuousDataCollection, SparseDataCollection]:
-        ...
+    @property
+    def pixel_impedance_global_offset(self) -> np.ndarray:
+        return self.pixel_impedance - self.global_baseline
+
+    @property
+    def pixel_baseline(self) -> np.ndarray:
+        return np.nanmin(self.pixel_impedance, axis=0)
+
+    @property
+    def pixel_impedance_individual_offset(self) -> np.ndarray:
+        return self.pixel_impedance - np.min(self.pixel_impedance, axis=0)
+
+    @property
+    def global_impedance(self) -> np.ndarray:
+        return np.nansum(self.pixel_impedance, axis=(1, 2))
 
 
 @dataclass(eq=False)
-class EITData_(EITData):
+class EITData_(EITData):  # noqa: N801
     vendor: Vendor = field(init=False)
```

Comment:
Yes.

`EITData.from_path()` requires `vendor` as argument, so it can choose the right class to load the file. 
`DraegerEITData.from_path()` does not allow `vendor` as argument, because the vendor is implicit in the choice of class. Of course the same is true for SentecEITData and TimpelEITData.
It is possible to override `from_path()` in each of the three subclasses, but that would not be DRY. 

`EITData_` removes `vendor` from the initializer with `field(init=False)`. 
Each subclass sets `vendor` to the proper value.
`EITData_` also changes the arguments required for `from_path()`.

I wasn't able to solve this without an intermediate class. Any other solution I came up with requires re-implementing `from_path()` in each of the three subclasses.

Reply:
I think EITData_ might better be named _EITData, to indicate it's for internal use only (rather than just a similarly-named copy of EITData).

Reply:
Thanks for the explanation. Agreed about the second comment. Let's leave it like this for now, and I will ask around if someone in the office has a more elegant solution to this.

Reply:
Opened #152 for this